app/models/authenticator/base.rb

```diff
@@ -200,8 +199,10 @@ def authenticate_with_http_basic(username, password, request = nil, options = {}
       [!!result, username]
     end
 
-    def lookup_by_identity(username, *_args)
-      case_insensitive_find_by_userid(username)
+    def lookup_by_identity(username, request: nil, lookup_scope: nil)
+      @lookup                         ||= {}
```

Comment:
> what is the lifespan of this cache?

Just for the duration of the request.  This method was called multiple times as part of the `Authenticator::Base#authenticate` method, so was hoping to avoid redundant lookups.  That said, I put it in a "hash" incase it was ever passed different args, and I didn't want to return an invalid cached result if it wasn't the same as was requested originally.